lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Context.java

```diff
@@ -61,6 +66,13 @@ public S3Context withCredentials(ConnectorIdentity identity)
                     identity.getExtraCredentials().get(EXTRA_CREDENTIALS_SESSION_TOKEN_PROPERTY)));
             return withCredentialsProviderOverride(credentialsProvider);
         }
+
+        if (identity.getExtraCredentials().containsKey(EXTRA_CREDENTIALS_SSEC_KEY)) {
+            byte[] decodedKey = Base64.getDecoder().decode(identity.getExtraCredentials().get(EXTRA_CREDENTIALS_SSEC_KEY));
+            verify(decodedKey.length == 32, "AES encryption key must be 256 bits long");
```

Comment:
Let's move the decode and validation to the record constructor using `checkArgument()`. The contract is that the record must be constructed with a valid key.

We can also construct the MD5 in the record constructor using the decoded key. Then we only need to decode the key once.